app/history_v2.go

```diff
@@ -31,10 +31,17 @@ func HistoryV2Handler(app *App) func(c echo.Context) error {
 		if !authenticated {
 			return c.String(echo.ErrUnauthorized.Code, echo.ErrUnauthorized.Message)
 		}
+
 		messages := make([]*models.MessageV2, 0)
+		allMessages := make([]*models.MessageV2, 0)
 		collection := app.Defaults.MongoMessagesCollection
-		messages = mongoclient.GetMessagesV2(c, topic, from, limit, collection)
+		allMessages = mongoclient.GetMessagesV2(c, topic, from, limit, collection)
```

Comment:
The overall idea works, but I have one concern here: by filtering in the `history_v2` handler, we need to replicate this logic on `history` (v1) too.

Looking into the code, I believe that we can filter directly blocked messages on `mongoclient.GetMessagesV2` to solve our problem, by adding on more filter here: https://github.com/topfreegames/mqtt-history/blob/master/mongoclient/get_messages.go#L45

As `mongoclient.GetMessages` uses internally `mongoclient.GetMessagesV2` this could solve the problem on replicating the logic, does it make sense?